doc/code/qml_pulse.rst

```diff
@@ -0,0 +1,215 @@
+qml.pulse
+=========
+
+.. automodule:: pennylane.pulse
+
+Pulse control
+-------------
```

Comment:
@lillian542 if you would like, you could also include this section inside the `pulse/__init__.py` file - it would render exactly the same (since it will be picked up by `automodule`. The advantage is that the description of the code lives closer to the code :)